### ISSUE TITLE
Strange error after CoffeeScript upgrade to 1.11

### ISSUE THREAD

**Issue by ukoloff** [docpad/docpad#1052]
My plugin [docpad-plugin-without](https://github.com/ukoloff/docpad-plugin-without) started to fail tests after CoffeeScript released v1.11.0 (on September 24th).

Error message is rather strange:
```
ReferenceError: docpad is not defined
  at DocPad.loadLocale (/home/travis/build/ukoloff/docpad-plugin-without/node_modules/docpad/out/lib/docpad.js:1601:7)
  at DocPad.<anonymous> (/home/travis/build/ukoloff/docpad-plugin-without/node_modules/docpad/out/lib/docpad.js:1574:29)
  at DocPad.getLocale (/home/travis/build/ukoloff/docpad-plugin-without/node_modules/docpad/out/lib/docpad.js:1577:10)
  at DocPad.module.exports.docpadUtil.action (/home/travis/build/ukoloff/docpad-plugin-without/node_modules/docpad/out/lib/util.js:332:17)
```
See [details](https://travis-ci.org/ukoloff/docpad-plugin-without/builds/162420061)

Manually pinning CoffeeScript version to 1.10 (inside docpad's package.json) seems to fix problem.

**Comment by ukoloff** [comment]
The same errors seems to appear even before plugin:

```
ReferenceError: docpad is not defined
  at DocPad.loadLocale (/home/travis/build/ukoloff/ukoloff.tk/node_modules/docpad/out/lib/docpad.js:1601:7)
  at DocPad.<anonymous> (/home/travis/build/ukoloff/ukoloff.tk/node_modules/docpad/out/lib/docpad.js:1574:29)
  at DocPad.getLocale (/home/travis/build/ukoloff/ukoloff.tk/node_modules/docpad/out/lib/docpad.js:1577:10)
```
[here](https://travis-ci.org/ukoloff/ukoloff.tk/jobs/162449483)

**Comment by ukoloff** [comment]
And finally I just did `docpad upgrade`, it installed CoffeeScript v.1.11 (inside itself) and now simple `docpad` command fails with the same error trace!

I deleted folder `C:\Users\Me\AppData\Roaming\npm\node_modules\docpad\node_modules\coffee-script` (docpad's personal copy of CoffeeScript v1.11) - `docpad` command started to work again!

File 'docpad/out/lib/docpad.js':
``` js
// Generated by CoffeeScript 1.10.0
var BasePlugin, CSON, Collection, DocPad, ...
```
is (pre)compiled with CoffeeScript v1.10. It declares DocPad and uses docpad (in 1601).

`Curiouser and curiouser!' cried Alice

**Comment by ukoloff** [comment]
This is probably [error in cson-parser](https://github.com/groupon/cson-parser/issues/56)...

**Comment by mikeumus** [comment]
Hi @ukoloff, thanks for reporting this issue. May you please provide some more information so we can help out. Specifically the items here from the docpad.org/docs/support "BUG REPORTS VIA GITHUB'S ISSUE TRACKER" page:

- Your DocPad version docpad --version
- Your Node.js version node --version
- Your npm version npm --version
- Your operating system's name, architecture, and version
- What you did
- What happened
- What you expected
- The skeleton that was affected (if applicable)
- The docpad-debug.log file when you run DocPad in debug mode by specifying the -d flag (e.g., docpad run -d)

Thanks.

**Comment by ukoloff** [comment]
I believe this information is irrelevant, but nevertheless:

  - **docpad --version** v6.78.6 (global installation: C:\Users\Me\AppData\Roaming\npm\node_modules\docpad)
  - **node --version** v4.4.2
  - **npm --version** 3.10.8
  - **os** Windows 7 Home 64bit
  - **Did** docpad upgrade
  - **Happened** docpad fails to start, with `ReferenceError: docpad is not defined` complaint
  - **Expected** docpad still working
  - **Skeleton** not applicable. Fails before any skeleton or plugin
  - **docpad -d**
```
C:\Users\Me\AppData\Roaming\npm\node_modules\docpad\out\lib\docpad.js:1601
      docpad.error(locale);
      ^

ReferenceError: docpad is not defined
  at DocPad.loadLocale (C:\Users\Me\AppData\Roaming\npm\node_modules\docpad\out\lib\docpad.js:1601:7)
  at DocPad.<anonymous> (C:\Users\Me\AppData\Roaming\npm\node_modules\docpad\out\lib\docpad.js:1574:29)
  at DocPad.getLocale (C:\Users\Me\AppData\Roaming\npm\node_modules\docpad\out\lib\docpad.js:1577:10)
  at new ConsoleInterface (C:\Users\Me\AppData\Roaming\npm\node_modules\docpad\out\lib\interfaces\console.js:63:21)
  at C:\Users\Me\AppData\Roaming\npm\node_modules\docpad\out\bin\docpad.js:35:12
  at new DocPad (C:\Users\Me\AppData\Roaming\npm\node_modules\docpad\out\lib\docpad.js:2140:9)
  at startDocPad (C:\Users\Me\AppData\Roaming\npm\node_modules\docpad\out\bin\docpad.js:29:10)
  at checkDocPad (C:\Users\Me\AppData\Roaming\npm\node_modules\docpad\out\bin\docpad.js:19:12)
  at Object.<anonymous> (C:\Users\Me\AppData\Roaming\npm\node_modules\docpad\out\bin\docpad.js:46:1)
  at Module._compile (module.js:409:26)
  at Object.Module._extensions..js (module.js:416:10)
  at Module.load (module.js:343:32)
  at Function.Module._load (module.js:300:12)
  at Module.require (module.js:353:17)
  at require (internal/module.js:12:17)
  at Object.<anonymous> (C:\Users\Me\AppData\Roaming\npm\node_modules\docpad\bin\docpad:2:1)
  at Module._compile (module.js:409:26)
  at Object.Module._extensions..js (module.js:416:10)
  at Module.load (module.js:343:32)
  at Function.Module._load (module.js:300:12)
  at Function.Module.runMain (module.js:441:10)
  at startup (node.js:139:18)
  at node.js:968:3
```

I believe that error is raised inside `CSON.parseCSONFile`, it is the issue of cson-parser.
Docpad catches this error, but in turn fails to display it in line 1601 (docpad.error not working).

**Comment by balupton** [comment]
Interesting.

Seems cson-parser has fixed this - https://github.com/groupon/cson-parser/issues/56#issuecomment-249394512 - does doing a `rm -Rf node_modules; npm cache clean; npm i` fix the problem?

**Comment by balupton** [comment]
https://github.com/docpad/docpad/commit/b2a634d7b73f7b0f72f9335d1d75c2a671f85cb7 should fix this, @mikeumus or @SteveMcArthur can you do a version bump, `nake prepare`, and `nake release` - I'm travelling right now

**Comment by balupton** [comment]
Re-opening until fixed version is published.

**Comment by ukoloff** [comment]
Yes, updating to new version of cson-parser does the trick, tested.

**Comment by balupton** [comment]
Released to v6.79.0
